firmware/include/caseTempController.h

```diff
@@ -0,0 +1,40 @@
+//*****************************************//
+// includes                                //
+//*****************************************//
+#include "caseTempController/config.h"
+#include "caseTempController/temperature.h"
+#include "caseTempController/fan.h"
+
+//*****************************************//
+// class                                   //
+//*****************************************//
+class CaseTempController
+{
+    private:
+
+        // a history of the last TAU temperature errors
+        int temperature_errors[TAU];
+
+        // marks the place of the oldest info in the temperature_errors array
+        int* p_oldest_temperature_error;
+
+        // temperature sensor
+        TempSensor tempSensor;
+
+        // fan
+        Fan fan;
+
+        // counter used to check if controller should start measuring/calculating
+        int sleep_check;
+
+    public:
+
+        CaseTempController()
+        {
+            this->sleep_check = 0;
```

Comment:
What does this do?

Reply:
I do not want to measure/adjust on every loop-through of the main file. Instead, I'm doing so on every 30000th loop in order to save some calculation time (and not trash your logs too much).

In the function readTemperatureAndAdjustFanSpeed (formerly known as caseTempController), the whole logic is encased by an "if (sleep_check % SLEEP == 0)", where SLEEP can be found in config.h and is 30000.